### DISCUSSION TITLE
Auto generete Links to pages from available files in pages/

### DISCUSSION THREAD

**Discussion by MariuzM** [vercel/next.js#19620] [Help] [answered]
As suggested on the ticket https://github.com/vercel/next.js/issues/19616 i have tried the example but getting empty array, any suggestions? 

`src/pages/page3.tsx`

```ts
import globby from 'globby'

const Page3 = () => {
  return (
    <>
      <h1>Page3</h1>
    </>
  )
}

export default Page3

export async function getStaticProps(context: any) {
  const pages = await globby(['pages/**/*{.tsx,.mdx}', '!pages/_*.js', '!pages/api'])

  console.log('🚀 ~ file: page3.tsx ~ line 53 ~ pages', pages)

  return {
    props: {},
  }
}
```
![Screen Shot 2020-11-28 at 09 23 43](https://user-images.githubusercontent.com/36937039/100498621-6cad1700-315b-11eb-90b1-703f7d74165c.png)

**Comment by Tomburgs** [comment, marked as answer]
You just have to specify the path from root.

e.g if I have a set-up like this:
```
.
├── next.config.js
├── package.json
└── src
    └── pages
```

You would have to use globby as follows:
```js
await globby(['src/pages/**/*{.tsx,.mdx}', '!src/pages/_*.js', '!src/pages/api'])
```

I would also like to point out that you could also use `self.__BUILD_MANIFEST` in runtime, as it will include all the pages that you have automatically.

**Reply by MariuzM** [reply]
Perfect the path was my issue,

I Am curious who do i use this `self.__BUILD_MANIFEST` ?

**Reply by Tomburgs** [reply]
Glad I cloud help. :)

What I meant is that `self.__BUILD_MANIFEST` also includes the built pages and the resources associated with them.

![__BUILD_MANIFEST](https://image.prntscr.com/image/ZWGCwRotQsC2ggTVM3kVTw.png)

**Reply by MariuzM** [reply]
Looks like self.__BUILD_MANIFEST does not really work in dev mode but in build yeah getting the pages
